nemo/core/optim/distributed_adam.py

```diff
@@ -284,8 +290,15 @@ def init_param_buffer(self) -> None:
             buffer_sizes[dtypes] = max(bucket.contiguous_buffer_offset + bucket.bucket_size, buffer_sizes[dtypes])
         for dtypes, buffer_size in buffer_sizes.items():
             _, _, param_sync_dtype = dtypes
-            self._param_buffers[dtypes] = torch.zeros([buffer_size], dtype=param_sync_dtype, device=self.device)
-
+            if self.nccl_ub:
```

Comment:
For backward compatibility:
```suggestion
            if getattr(self, "nccl_ub", False):
```